app/views/play/CampaignView.js

```diff
@@ -318,59 +318,61 @@ module.exports = (CampaignView = (function () {
               })
               )
               return this.listenToOnce(this.courseLevels, 'sync', () => {
-                let idx, k, v
-                const existing = this.campaign.get('levels')
-                const courseLevels = this.courseLevels.toArray()
-                const classroomCourse = _.find(globalVar.currentView.classroom.get('courses'), { _id: globalVar.currentView.course.id })
-                const levelPositions = {}
-                for (const level of Array.from(classroomCourse.levels)) {
-                  if (level.position) { levelPositions[level.original] = level.position }
-                }
-                for (k in courseLevels) {
-                  v = courseLevels[k]
-                  idx = v.get('original')
-                  if (!existing[idx]) {
-                    // a level which has been removed from the campaign but is saved in the course
-                    this.courseLevelsFake[idx] = v.toJSON()
-                  } else {
-                    this.courseLevelsFake[idx] = existing[idx]
-                    // carry over positions stored in course, if there are any
-                    if (levelPositions[idx]) {
-                      this.courseLevelsFake[idx].position = levelPositions[idx]
-                    }
+                return this.listenToOnce(this.campaign, 'sync', () => {
```

Comment:
i don't think this could work. since courseLevels and campaign load is sync and we're not sure which is sync first. (actually I doubt campaign would be fast)  so I think you need some thing like 
```
listenToOnce(courseLevels, () => courseLevelsLoaded = true; checking())
listenToOnce(campaign, () => campaignLoaded = true; checking()) 
```
and in checking function, 
only when courseLevelsLoaded and campaignLoaded both true, run following logics, reasonable?